### ISSUE TITLE
`builtins.merge` crashes on a new codebase with `UnknownText "Effect"`

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#2719]
```
arya@Halp rehash-codebase % stack exec unison -- --codebase-create /tmp/cc --no-base
I backed up your codebase to /tmp/cc/.unison/v2/unison.sqlite3.1638651932
Need: BranchE (ObjectId 1)
Done: BranchE (ObjectId 1)
Done: CausalE CausalHashId (HashId 1)
Finished migrating, initiating cleanup.

  I created a new codebase for you at /private/tmp/cc

  Now starting the Unison Codebase Manager (UCM)...


   _____     _             
  |  |  |___|_|___ ___ ___ 
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|
  
  👋 Welcome to Unison!
  
  You are running version: latest-675-g8d3d89649'


  Get started:
  
    📖 Type help to list all commands, or help <cmd> to view help for one command
    🎨 Type ui to open the Codebase UI in your default browser
    📚 Read the official docs at https://unisonweb.org/docs
    🌍 Visit Unison Share at https://share.unison-lang.org to discover libraries
    👀 I'm watching for changes to .u files under /Users/arya/work/unison/rehash-codebase

.> builtins.merge
Exception: UnknownText "Effect"
CallStack (from HasCallStack):
  error, called at src/Unison/Codebase/SqliteCodebase.hs:1014:29 in unison-parser-typechecker-0.0.0-3LFNPp4mFYO32cmeBkPWEL:Unison.Codebase.SqliteCodebase
unison: UnknownText "Effect"
CallStack (from HasCallStack):
  error, called at src/Unison/Codebase/SqliteCodebase.hs:1014:29 in unison-parser-typechecker-0.0.0-3LFNPp4mFYO32cmeBkPWEL:Unison.Codebase.SqliteCodebase
arya@Halp rehash-codebase % 
```

**Comment by aryairani** [comment]
`git bisect` says this started happening somewhere between _ and _. (I'm two days into git bisect... no answer yet)

**Comment by aryairani** [comment]
I finally "finished" a very unsatisfying 2-day long `git bisect`:

```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
8f380766fe9dbbba97c1383558da0ca91d3a3415
0e22887f15c9fecb1b5d5706fe77652547f624b4
6344a417c882d3a994beaab06ab18b336c5166a0
06a50aed5b716f472e39dfbe33e519ff7e23f670
275a8825d6ce0749e7490ee3c55ee0c81e7d69c1
360cbba4c71b219c413934cceb94bb4a7fdff504
a872d792bf5a737043a769f36b3c714e24669163
76fbb320cfbb277c33c9084428d0f53744127bb0
7bb6d8825ac20867af6480b937cad36fbe7196ab
842aa4a0bd5101a49210e8e6ad83f43b5a917323
9481a9c444102124d4de663c9593ee3eef672ba2
9cb6c0c33726639a329146cbe03e70da15e43d41
2a15cfd9a4d65aba4de7cc3d8afc4af996c16148
303ba0faef6b89520afc1147336383373cb94138
94fcea6eeacf5ba3e25a5b477cad9c68b5b9ee42
2ed8e687adf2b2b4443add0545c2ba9a89c9e3ad
97acf505f8ca7b79eb43e8e05a81a492d6c1e17c
4e332b7454006b6503be794827ca5825a29d88d9
639b3d44ce60b27d8d929ffccb4f8880b4564c5d
33000db21f05a788022946d7e517ee4a70bee9b3
fe52ef99b78911c95e19bf80f73435153258483c
7a229fb2808ad59692ff942ada07501e4f74ce53
ba9271ac2810f647dfe785d7809f2ce83d450240
d4de7fa0a12cb4b61be004bc7362fcc7232033ba
07fb8343ac6ec0339d270c643bd67ccd2edf4f0a
3b5cbd78888f3065690ce897e04de45c9bd2a74d
d9f5c7b021120b05d53c8fe1b60213e1f2f31112
43a41106f3ee28ed3622bce7098ccf75961c4414
4a7b0f560495be3ec45751f9690bab8582ba059a
a435674efccd13bda4ddc261fa865b5d4ee72575
1a6b417544369c3ed82ee66f4109d54a16e41aea
20f18c55595529ee0b4ad945d9b7f8c0d0bc7cd1
c40b41bc072e5714317edc8e5bd2466bddb40c79
012e8aa3d9b9cd4eec1e995f24006bea193fe794
e619802bc0f0f3f576f11eae4b972e0a7d390364
0a5a01b31eeca4b4dc8c9f0c7bb1c4e30b8f15c7
d2d619d897d82bfe2d7b454a7950526331fe190b
1020290a8e3d828e7ff197df9b28c55b34c6b443
7c42120d0c8f9f946b8bc27a56d32362d088bd07
87a8057710f56fcd7aea9e7c9eb71f317531a081
0f87f184cbac3602e38001c401842c306cec94ac
35c813864e5c563d0b5886291142d9a5f07769fd
c75175a6a8f581008b60b4f716c666014995344f
8c171528bd89ddaf43b54c5eec197b5623014cd9
9d56b5ee17fa35ebc6629279ad6ddeddb3822aed
66eafa927c7ea0a14034a6b091e7c595f73849ce
93437c7631c7435e785f177cf87466a5da66ade6
98e0e6be1b02d788e50b93047e025dca858838a1
d761a5eb2c7486a6a725f58ca02890756685e1b5
502b89f6d5d80a386e36ac830206eabf4fb4cd3c
a30b932cc6ed14e164036840e8cdfd6d2cfff708
931e4f606f1f491b341550af08f7c273e0c2a049
f30bbafa0cf71a1ec64ead0f362c7ba5c05468c2
7a990632c19ff9623470337472680e992534de0a
9dd25091a8348a7a5466113cdb5ff0a105f85c12
6ac566f625701466cfe50ff495b69ba32197ecdf
767b08fc48a1a7026004a2a18cdcdb268fbfd9f1
d354cf99f7f1298284c9a9ad05640ded7b80ba21
2f12a34df8eb733645711a0549a907530a3e1961
9724bc4ea56dfedb22b347203ae33fea3daf8def
e9e39726899759d124495df59fe9833d884178f6
3946e25e8cf8fdf369647baacc0fdb796f12e959
4114ae5bf4fa93ad3048027e9baeb71f686c560e
0640f62152fed88f1a421ecb5cfcdd1d4d211052
7cb1ab306e8910e25e74f290de1daa5be7d31229
5990ab240c95eea0cbbbdd326aadadb18d3a84bc
e013511dd86b0462fd9937f5c394aebae27892c9
6c07b36b926c47b0748fb00b4301ae5df54bcb75
e1bd091b77a2c01dc9662c6d69c07b036dc6ef8a
5f495f0147feee0161a27b3280cbc77e7ffba8a7
8d47300f0e259f71944f0878197257f5e9ed4b00
9b898c6188753527f84e85e57b6ad1437359d2f6
d5fcd214ebb3e0ba44dab2c2b5e8e397b3b73602
482f9c24be886d7819953297837ca90efa066077
6f080b2eb61985fd8b5b61b87bf21a45da02ba88
2384646468281727d14cc196fa2453fc906ba848
e742e8cc1a33aa285ad70761313ae81fdd1373c6
2a1f537461a2a611718613f5e9a0a99e0d2dc315
6d39ecf8dde2ebccb45ab5e9e472641fb2331f50
0497f7dc36d0946ba2d51c13883830663e1eb651
6ee3b40f170d67aa995db4156d71ca22de45f9b4
6d5ac667a5fae8907b59463ea70c1263d5a0f5b6
9ceae3eb0fc198536d0ed56a290a7e78a2408858
20146d233fc174bab02abcab2a1231008de17c77
We cannot bisect more!
```

**Comment by aryairani** [comment]
fixed with #2730
